tool/src/version_builder.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:build/build.dart';
+import 'package:pub_semver/pub_semver.dart';
+import 'package:yaml/yaml.dart';
+
+class PackageVersionBulder extends Builder {
+  @override
+  Future build(BuildStep buildStep) async {
+    var content =
+        await buildStep.readAsString(new AssetId.parse('pana|pubspec.yaml'));
```

Comment:
I _guess_ this is fine since we don't expect it to be used outside of this package... but it seems weird to hardcode it when we have the information anyway.

Reply:
Explain?

Reply:
by hardcoding `pana` here you're writing a builder that will only ever work in the pana package. That's not the _only_ reason it only works here (the fact that its in `tool/` rather than `lib/` is the other reason), but it will mean there are more changes we'd need to make if we wanted to move this somewhere else and share it.

The way we'd write this if we wanted it to be general purpose instead of hardcoded is to check `buildStep.inputId.package` instead of `pana` and I don't see that as any more complicated than the current code so I don't see value in leaving it hardcoded.

Reply:
Fixed. It's good to show best practices, me thinks.